gensim/models/word2vec_inner.pyx

```diff
@@ -633,6 +632,9 @@ def train_sentence_sg(model, sentence, alpha, _work):
             else:
                 codelens[i] = 1
             result += 1
+    # single randint() call avoids a big thread-sync slowdown
+    for i, item in enumerate(np.random.randint(0, window, sentence_len)):
+        reduced_windows[i] = item
```

Comment:
I hadn't even thought of that, being in C-thinking mode when I wrote it (and filling the `cdef np.uint32_t reduced_windows[MAX_SENTENCE_LEN]` array). But I just tried…

`reduced_windows[:sentence_len] = np.random.randint(0, window, sentence_len)`

…and got a couple cython errors. On the other hand, this variant…

`reduced_windows[:sentence_len] = <np.uint32_t *>(np.PyArray_DATA(np.random.randint(0, window, sentence_len)))`

…will compile, so I'm giving it a try now.